client/router_update.go

```diff
@@ -95,6 +96,42 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 	return updateRouter || updateController || updateSite, nil
 }
 
+func (cli *VanClient) RouterUpdateLogging(ctx context.Context, settings *corev1.ConfigMap, hup bool) (bool, error) {
+	siteConfig, err := cli.SiteConfigInspect(ctx, settings)
+	if err != nil {
+		return false, err
+	}
+	configmap, err := cli.KubeClient.CoreV1().ConfigMaps(settings.ObjectMeta.Namespace).Get(types.TransportConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	routerConfig, err := qdr.GetRouterConfigFromConfigMap(configmap)
+	if err != nil {
+		return false, err
+	}
+	updated := configureRouterLogging(routerConfig, siteConfig.Spec.RouterLogging)
+	if updated {
+		routerConfig.WriteToConfigMap(configmap)
+		_, err = cli.KubeClient.CoreV1().ConfigMaps(settings.ObjectMeta.Namespace).Update(configmap)
+		if err != nil {
+			return false, err
+		}
+		if hup {
+			router, err := cli.KubeClient.AppsV1().Deployments(settings.ObjectMeta.Namespace).Get(types.TransportDeploymentName, metav1.GetOptions{})
```

Comment:
@grs is this update going to recycle the router pod?
Is it worth doing it live through management API instead, to avoid recycling?

Reply:
I see that as an optimisation/enhancement. I'd rather wait and see if we find that changing the log levels is common enough that avoidign restarts is worth it.